src/kmer_collection.h

```diff
@@ -126,57 +153,76 @@ class KmerCollection {
     m_scoreMin(std::move(other.m_scoreMin)),
     m_kmerListSize(std::move(other.m_kmerListSize)),
     m_kmerList(std::move(other.m_kmerList)),
-    m_usedGenerator(&KmerCollection::generateKmersN),
+    m_usedGenerator(&KmerCollection::generateKmerN),
     m_compress(std::move(other.m_compress))
-  {}
+  {
+    other.m_kmerList = nullptr;
+  }
 
-  virtual ~KmerCollection() {
-    if(m_kmerList) {
-      delete[] m_kmerList;
-    }
+  /**
+  * Generates kmers if necessary.
+  * \return Number of kmers.
+  */
+  size_t size() {
+    generateKmers();
+    return m_kmerListSize;
   }
 
+  /**
+  * Generates kmers if necessary.
+  * \return Iterator to KmerCollection's first element.
+  */
   KmerCollection::iterator begin() {
+    generateKmers();
+    return KmerIterator(this, 0);
+  }
+
+  /**
+  * \return Iterator to the element after KmerCollection's last element.
+  */
+  inline KmerCollection::iterator end() {
+    return KmerIterator(this, m_kmerListSize);
```

Comment:
What if `m_kmerListSize` is still 0 when calling `end()`?
Shouldn't we also `generateKmers()` here?